HW6.py

```diff
@@ -0,0 +1,224 @@
+            print('{} can\'t eat product from refrigerator!'.format(self.name))
+        else:
+            print('{} eat {}'.format(self.name, product.name))
+            del product
+
+
+class Parent(User):
+    """Parent is User who can call to master, pay and earn money"""
+    money = 1400
+
+    def call_to_master(self, master, refr):
+        """Using this method parent can call to master.
+
+        After call master will try to fix refrigerator using master.try_to_fix
+        method. This method returns result (True o False), and this result will
+        be printed
+        """
+        job_result = master.try_to_fix(self, refr)
+        print('{} called to {}.'.format(self.name, master.name), end=' ')
+        print('He fixed it' if job_result else 'He can\'t fix it')
+
+    def earn_money(self, salary=200):
+        """Parent get some money after this method"""
+        self.money += salary
+
+
+class Child(User):
+    """Child is user who can hit refrigerator"""
+    def hit_fridge(self, refr):
+        """After hit refriger. will be broken with a thirty percent chance """
+        if random.random() < 0.3:
+            refr.broken = True
+            print('{} broke {}!'.format(self.name, refr.name))
+        else:
+            print('{} hit {}, but not broke.'.format(self.name, refr.name))
+
+
+class Master(object):
+    """Master is a person who can fix refrigerators if he have enough skill"""
+    money = 100
+
+    def __init__(self, skill, name='Master', company='own'):
+        self.skill = skill
+        self.name = name
+        self.company = company
+
+    def advertisement(self):
+        """Advertising of master. Skills always increase by two"""
+        print('Highly skilled master {0} from {1} company is looking for work!'
+              ' My skill is {2}. Prices:\nImport refrigerator: {3}\nDomestic '
+              'refrigerator: {4}'.format(self.name, self.company,
+                                         str(self.skill + 2),
+                                         str(self.skill * 150),
+                                         str(self.skill * 100)))
+
+    def try_to_fix(self, customer, refr):
+        """Master will try to fix refrigerator
+
+        Price formula: skill * 100 * сoeff
+        Netherlands refrigerators can be fixed by only masters with >=7 skill,
+        coeff is 1,5
+        Belarus refrigerators can be fixed only by masters with >=3 skill,
+        coeff is 1, and also help to improve some skills.
+        Payment is made by the client (Parent)
+        Сonsultation is free.
+        """
+        if refr.made_in == 'The Netherlands':
+            if self.skill >= 7:
+                refr.broken = False
+                customer.money -= self.skill * 100 * 1.5
+                return True
+        elif refr.made_in == 'Belarus':
+            if self.skill >= 3:
+                refr.broken = False
+                customer.money -= self.skill * 100 * 1
+                self.skill += 0.5
```

Comment:
he can unlimited improve his skill repairing atlants and can't level up when repair netherlands.

Reply:
When I invented I mean that Belorussian refrigerators are so bad, that every time they have new problem and master can improve his skills solving it :)
But I agree that is is not logical. Now fixed: Netherlands refrigerators also help to level up. And skills limit has been set.

Reply:
I have new Atlant fridge and really like it ^)